src/test/java/randoop/main/minimizer/MinimizerTests.java

```diff
@@ -0,0 +1,196 @@
+   * @throws IOException thrown if output or expected output files can't be read
+   */
+  private void testWithInput(String filename) throws IOException {
+    testWithInput(testDir + filename, null, 30);
+  }
+
+  /**
+   * Test the minimizer with an input file.
+   *
+   * @param inputFilePath input file to minimize: a JUnit test suite
+   * @param dependencies dependencies needed to compile and run the input file. This parameter is an
+   *     array of Strings, each representing an element of the classpath, for instance, a directory
+   *     or a jar file.
+   * @param timeoutLimit maximum number of seconds allowed for any test case within the input file
+   *     to run
+   * @throws IOException thrown if output or expected output files can't be read
+   */
+  private void testWithInput(String inputFilePath, String[] dependencies, int timeoutLimit)
+      throws IOException {
+    String outputFilePath =
+        new StringBuilder(inputFilePath)
+            .insert(inputFilePath.lastIndexOf('.'), "Minimized")
+            .toString();
+    String expectedFilePath = inputFilePath + ".expected";
+
+    // Obtain File object references
+    File inputFile = new File(inputFilePath);
+    File outputFile = new File(outputFilePath);
+
+    // Obtain the complete path to the input, output, and expected files
+    inputFilePath = inputFile.getAbsolutePath();
+    outputFilePath = outputFile.getAbsolutePath();
+
+    String classPath = null;
+    if (dependencies != null) {
+      classPath = "";
+      for (String s : dependencies) {
+        File file = new File(s);
+        classPath += (pathSeparator + file.getAbsolutePath());
+      }
+    }
+
+    // Create the arguments array and invoke the minimizer
+    Minimize.mainMinimize(inputFilePath, classPath, timeoutLimit);
+
+    // Compare obtained and expected output
+    String obtainedOutput = readFile(outputFilePath, Charset.defaultCharset());
+    String expectedOutput = readFile(expectedFilePath, Charset.defaultCharset());
+
+    // Handle carriage returns on different OSes.
+    expectedOutput = expectedOutput.replaceAll("\\r\\n", "\n");
+    obtainedOutput = obtainedOutput.replaceAll("\\r\\n", "\n");
+
+    assertEquals(expectedOutput, obtainedOutput);
+  }
+
+  /**
+   * Read a file and return the contents as a String.
+   *
+   * @param filePath path to file
+   * @param encoding character encoding
+   * @return contents of the input file
+   * @throws IOException if the file can't be read
+   */
+  private static String readFile(String filePath, Charset encoding) throws IOException {
+    byte[] contents = Files.readAllBytes(Paths.get(filePath));
+    return new String(contents, encoding);
+  }
+
+  @Test
+  public void test1() throws IOException {
+    testWithInput("TestInput1.java");
+  }
+
+  @Test
+  public void test2() throws IOException {
+    testWithInput("TestInput2.java");
+    // Test input contains while loop.
```

Comment:
Comments like these may be better in the files themselves, where they will be seen by someone who is debugging test failures.